packages/components/checkbox/src/composables/use-checkbox-status.ts

```diff
@@ -41,7 +41,7 @@ export const useCheckboxStatus = (
   const checkboxSize = useFormSize(computed(() => checkboxGroup?.size?.value))
 
   const hasOwnLabel = computed<boolean>(() => {
-    return !!(slots.default || props.label)
+    return !!slots.default || (!isNil(props.label) && props.label !== false)
```

Comment:
```suggestion
    return !!slots.default || !isNil(props.label)
```

The same issues exists when the value is false

```vue
<el-checkbox :label="false" />
```

Reply:
According to the current logic, if the value of `label` is `false`, it should not be displayed.
Initially, I also modified it according to your suggestion, but this will cause the tree test to fail.
https://github.com/element-plus/element-plus/actions/runs/5908255483/job/16028234409

Reply:
```diff
label: {
    type: [String, Boolean, Number, Object],
+    default: undefined,
  },
```

It seems that the problem is caused by the default value of label being false rather than undefined

Reply:
Even if the default value is set to `undefined`, when the label value is set to `false`, the page will display the string false, which seems unreasonable.

Reply:
`label=false` -> the label has a value -> `hasOwnLabel = true`

 show the false should be a reasonable behavior.

Reply:
I think it's better to be consistent with the previous behavior, otherwise, it will be a `breaking change`,

Reply:
I think it can be modified during this major version update

Reply:
Ok.

Reply:
I think it's possible to modify the title of PR so that the default value has changed in the changelog

Reply:
Updated.

Reply:
How about this?

refactor(components): [checkbox] set the label default value to undefined